File: 78.subset.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> q;
        q.push_back(vector<int>());

        for (int i = 0; i < nums.size(); i++) {
            int size = q.size();
            for (int j = 0; j < size; j++) {
                vector<int> tmp = q[j];
                tmp.push_back(nums[i]);
                q.push_back(tmp);
            }
        }

        return q;
    }
};

File: 78.subset.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)

        def helper(i, res):
            if i >= n:
                return res
            resL = len(res)
            for j in range(resL):
                each = res[j]
                tmp = each[:]
                tmp.append(nums[i])
                res.append(tmp)
            return helper(i+1, res)

        return helper(0, [[]])